### PR TITLE
FIX-#2380: don't ignore lengths parameter for dask engine

### LINKED ISSUE #2380: Dask engine ignore lengths parameter

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

* Add `lengths` parameter handling for dask engine.
* Make `split_result_of_axis_func_pandas` always return a list value and removes a bug in which the wrong number of partitions is returned (in the case when `num_splits==1` and `lengths != 1`)

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/CONTRIBUTING.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2380 
- [ ] tests added and passing

### CODE REVIEW

File: modin/engines/dask/pandas_on_dask/frame/axis_partition.py [hunk 1]
```diff
@@ -68,8 +70,6 @@ def deploy_func_between_two_axis_partitions(
             *partitions,
             pure=False,
         )
-        if num_splits == 1:
```
Comment (YarShev) [inline modin/engines/dask/pandas_on_dask/frame/axis_partition.py:71]: Why is this removed? We have the code that handles returned not list values (see [here](https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/axis_partition.py#L94)). Also, why is not `lengths` considered here?

File: modin/engines/dask/pandas_on_dask/frame/axis_partition.py [hunk 2]
```diff
@@ -43,13 +43,15 @@ def deploy_axis_func(
...
```
Comment (YarShev) [inline modin/engines/dask/pandas_on_dask/frame/axis_partition.py:46]: The same comment as for `deploy_func_between_two_axis_partitions`.

File: modin/data_management/utils.py [hunk 3]
```diff
@@ -82,15 +82,16 @@ def split_result_of_axis_func_pandas(axis, num_splits, result, length_list=None)
...
     if length_list is not None:
         length_list.insert(0, 0)
         sums = np.cumsum(length_list)
         if axis == 0:
             return [result.iloc[sums[i] : sums[i + 1]] for i in range(len(sums) - 1)]
         else:
             return [result.iloc[:, sums[i] : sums[i + 1]] for i in range(len(sums) - 1)]
+
+    if num_splits == 1:
+        return [result]
```
Comment (YarShev) [inline modin/data_management/utils.py:94]: Why do we wrap to a list?

**Review by YarShev — COMMENTED** [review]
Also, I think we need to move [this logic](https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/axis_partition.py#L219) into `split_result_of_axis_func_pandas` with corresponding handling.

File: modin/data_management/utils.py (hunk 3 above)
Comment (anmyachev) [reply modin/data_management/utils.py:94]: This is a simplification - now `split_result_of_axis_func_pandas` function returns `list` everywhere (as it is written in the docstring), which in turn localizes this edge case in this function, and not in several.

File: modin/engines/dask/pandas_on_dask/frame/axis_partition.py (hunk 1 above)
Comment (anmyachev) [reply modin/engines/dask/pandas_on_dask/frame/axis_partition.py:71]: Now there is no case when the type is different from the `list`.(Simplification)

File: modin/engines/dask/pandas_on_dask/frame/axis_partition.py (hunk 2 above)
Comment (anmyachev) [reply modin/engines/dask/pandas_on_dask/frame/axis_partition.py:46]: Answered above.

File: modin/engines/dask/pandas_on_dask/frame/axis_partition.py (hunk 1 above)
Comment (anmyachev) [reply modin/engines/dask/pandas_on_dask/frame/axis_partition.py:71]: The tests did not fall - so I did not pay attention.

I will see how to properly process `lengths`.

Comment (anmyachev) [reply modin/engines/dask/pandas_on_dask/frame/axis_partition.py:71]: I think there is no reason to add length handling to `deploy_func_between_two_axis_partitions` because two frames will be combined into one by some binary operation.

**Comment by anmyachev** [comment]
@YarShev ready for review

File: modin/engines/dask/pandas_on_dask/frame/axis_partition.py (hunk 1 above)
Comment (YarShev) [reply modin/engines/dask/pandas_on_dask/frame/axis_partition.py:71]: I see the point. There is no reason.

**Comment by YarShev** [comment]
@anmyachev , apparently, now we don't have a case when return value is different from `list`. So, I think we need to remove this [check](https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/axis_partition.py#L94) and keep returning `list`.

**Comment by YarShev** [comment]
Also, can you rebase on current master to fix failing of Omnisci tests?

**Review by YarShev — APPROVED** [review]
@anmyachev , thanks, LGTM!
